src/pubsub/ua_pubsub_manager.h

```diff
@@ -16,12 +16,24 @@ _UA_BEGIN_DECLS
 
 #ifdef UA_ENABLE_PUBSUB /* conditional compilation */
 
+#ifdef UA_ENABLE_PUBSUB_EVENTS
+typedef struct PublishedDataSetEventEntry {
+    UA_PublishedDataSet *pds;
```

Comment:
It isn't needed. We added it, to improve runtime. The PublishedDataSet can't be accessed directly from a DSW, it only offers the NodeId of a PDS. The algorithm to find the PDS (UA_PublishedDataSet_findPDSbyId) iterates over all PDSs, this would affect the efficiency.